lib/AST/Module.cpp

```diff
@@ -1247,6 +1247,44 @@ bool ModuleDecl::isSystemModule() const {
   return false;
 }
 
+namespace {
+enum class SerializedModuleKind {
+  Other,
+  Clang,
+  Swift,
+};
+}
+static SerializedModuleKind getSerializedModuleKind(const ModuleDecl *M) {
+  Optional<SerializedModuleKind> K;
+
+  for (auto File : M->getFiles()) {
+    switch (File->getKind()) {
+      case FileUnitKind::Source:
+      case FileUnitKind::Builtin:
+      case FileUnitKind::Derived:
+        return SerializedModuleKind::Other;
+      case FileUnitKind::SerializedAST:
+        assert(!K.hasValue() && "cannot handle multi-file modules");
```

Comment:
Today, yes. I don't want the system to start assuming this about Swift modules, though.

(The Clang one is fine. At one point I considered having overlays in the same Module as the ClangModuleUnit, but we actually make use of the shadowing behavior we get from today's model.)